rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -685,11 +749,351 @@ static std::string get_type_name(const char * type_support_identifier, void * ty
   }
 }
 
+
+extern "C" dds_dynamic_type_descriptor_t get_dynamic_type_descriptor_prim(
+  dds_dynamic_type_kind_t kind, const char *name, uint32_t num_bounds, const uint32_t *bounds, dds_dynamic_type_kind_t type)
+{
+  return{kind,name,{},{},num_bounds,bounds,DDS_DYNAMIC_TYPE_SPEC_PRIM(type),{},};
```

Comment:
I wonder if there is no better way to do this. The C example/test code uses designated initializers to set the fields that are required (and have the rest auto-initialized to 0), if that's still not supported in C++ perhaps doing:
```C++
dds_dynamic_type_descriptor t desc{};
desc.kind = kind;
...
```
works out better. I'm not quite sure, but writing it out as a struct initializer like this looks fragile.

Reply:
Accepted your suggestions.